posthog/api/search.py

```diff
@@ -0,0 +1,77 @@
+import re
+from typing import Any
+from django.contrib.postgres.search import SearchQuery, SearchRank, SearchVector
+from django.db.models import Model, Value, CharField, F, QuerySet
+from django.db.models.functions import Cast
+from django.http import HttpResponse
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+from posthog.models import Action, Cohort, Insight, Dashboard, FeatureFlag, Experiment, Team
+
+
+class SearchViewSet(StructuredViewSetMixin, viewsets.ViewSet):
+    permission_classes = [IsAuthenticated, ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission]
+
+    def list(self, request: Request, **kw) -> HttpResponse:
+        query = process_query(request.GET.get("q", "").strip())
```

Comment:
Does it make sense to search with no search query? Could raise a `ValidationError` here